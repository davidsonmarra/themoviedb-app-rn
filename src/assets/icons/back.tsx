import React from 'react';
import {Svg, Path} from 'react-native-svg';
import {colors} from '../../configs/theme';

const BackIcon = ({width = '24', height = '24', fill = colors.buttonText}) => {
  return (
    <Svg width={width} height={height} fill={fill} viewBox="0 -960 960 960">
      <Path d="M359-242L120-481l239-239 43 43-166 166h604v60H236l166 166-43 43z" />
    </Svg>
  );
};

export default BackIcon;
