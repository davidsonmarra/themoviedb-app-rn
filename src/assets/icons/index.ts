export {default as BackIcon} from './back';
export {default as SearchIcon} from './search';
