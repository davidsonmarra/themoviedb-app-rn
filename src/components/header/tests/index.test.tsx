import React from 'react';
import {TouchableOpacity} from 'react-native';
import Header, {Props} from '..';
import {Text} from '../..';
import {render, screen} from '@testing-library/react-native';
import {SearchIcon} from '../../../assets/icons';

const mockText = 'Snapshot test!';
const createComponent = ({text = mockText, onBack, ...rest}: Props) =>
  render(<Header text={text} onBack={onBack} {...rest} />);

const mockGoBack = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({goBack: mockGoBack}),
}));

describe('Button', () => {
  it('renders correctly', () => {
    const tree = createComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const instance = createComponent({}).root;
    expect(instance.findByType(Text)).toBeTruthy();
  });

  it('should call goBack', () => {
    createComponent({onBack: mockGoBack}).root;
    const backButton = screen.getByTestId('back-button');
    backButton.props.onClick();
    expect(mockGoBack).toHaveBeenCalled();
  });

  it('should call onBack when press back icon', () => {
    const instance = createComponent({onBack: mockGoBack}).root;
    const backIcon = instance.findAllByType(TouchableOpacity)[0];
    backIcon.props.onPress();
    expect(mockGoBack).toHaveBeenCalled();
  });

  it('should call rightIconAction when press right icon', () => {
    const mockRightIconAction = jest.fn();
    createComponent({
      rightIcon: () => <SearchIcon />,
      rightIconAction: mockRightIconAction,
    });
    const rightIcon = screen.getByTestId('right-buton-icon');
    rightIcon.props.onClick();
    expect(mockRightIconAction).toHaveBeenCalled();
  });
});
