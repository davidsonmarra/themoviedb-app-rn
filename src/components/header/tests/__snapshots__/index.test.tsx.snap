// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#01B4A4",
      "flexDirection": "row",
      "paddingHorizontal": 12,
      "paddingVertical": 16,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": 24,
      }
    }
    testID="back-button"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="#FFFFFF"
      focusable={false}
      height="24"
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={960}
      vbWidth={960}
      width="24"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M359-242L120-481l239-239 43 43-166 166h604v60H236l166 166-43 43z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "width": 12,
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "bold",
        },
        {
          "flex": 1,
          "textAlign": "center",
        },
      ]
    }
  >
    Snapshot test!
  </Text>
  <View
    style={
      {
        "width": 12,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": 24,
      }
    }
    testID="right-buton-icon"
  />
</View>
`;
